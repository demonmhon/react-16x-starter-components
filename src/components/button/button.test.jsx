import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import Button from './index';

describe('Button component', () => {
  test('it matches the snapshot', () => {
    const component = create(<Button />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('has default button text', () => {
    const component = shallow(<Button />);
    expect(component.find('.button__label').text()).toEqual('Button');
  });

  test('use the given childen as button text', () => {
    const buttonText = 'OK';
    const component = shallow(<Button>{buttonText}</Button>);
    expect(component.find('.button__label').text()).toEqual(buttonText);
  });

  test('has the provided class name', () => {
    const className = 'aaa';
    const component = shallow(<Button className={className} />);
    expect(component.hasClass(className)).toBe(true);
  });
});
