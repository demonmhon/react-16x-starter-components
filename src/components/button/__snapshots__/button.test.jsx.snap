// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component it matches the snapshot 1`] = `
<button
  className="button button--type-button"
  onClick={[Function]}
>
  <span
    className="button__label"
  >
    Button
  </span>
</button>
`;
